src/modules/helpchan.ts

```diff
@@ -81,13 +84,68 @@ export class HelpChanModule extends Module {
 		return `${this.CHANNEL_PREFIX}${decidedChannel}`;
 	}
 
+	private getOngoingChannels() {
+		return this.client.channels.cache
+			.filter(
+				channel =>
+					(channel as TextChannel).parentID === categories.ongoing,
+			)
+			.array() as TextChannel[];
+	}
+
 	@listener({ event: 'ready' })
 	async startDormantLoop() {
 		setInterval(() => {
 			this.checkDormantPossibilities();
 		}, dormantChannelLoop);
 	}
 
+	@listener({ event: 'ready' })
+	async initialCheckEmptyOngoing() {
+		for (const channel of this.getOngoingChannels()) {
+			if (await this.checkEmptyOngoing(channel)) {
+				await this.startEmptyTimeout(channel);
+			}
+		}
+	}
+
+	// Utility function used to check if there are no messages in an ongoing channel, meaning the bot
+	// is the most recent message. This will be caused if somebody deletes their message after they
+	// claim a channel.
+	async checkEmptyOngoing(channel: TextChannel) {
+		const messages = await channel.messages.fetch();
+
+		return messages.array()[0].author.id === this.client.user?.id;
+	}
+
+	async startEmptyTimeout(channel: TextChannel) {
+		if (this.ongoingEmptyTimeouts.has(channel.id)) return;
+
+		this.ongoingEmptyTimeouts.add(channel.id);
+
+		setTimeout(async () => {
+			this.ongoingEmptyTimeouts.delete(channel.id);
+
+			if (await this.checkEmptyOngoing(channel)) {
+				await this.markChannelAsDormant(channel);
+			}
+		}, ongoingEmptyTimeout);
+	}
+
+	@listener({ event: 'messageDelete' })
+	async onMessageDeleted(msg: Message) {
+		if (
+			msg.channel.type !== 'text' ||
+			!msg.channel.parentID ||
+			msg.channel.parentID !== categories.ongoing
+		)
+			return;
+
+		if (await this.checkEmptyOngoing(msg.channel)) {
```

Comment:
Remove this check, it just duplicates the check after the timeout, making rate-limiting more of a concern

Reply:
Done